lib/routes/utils/request_helper.js

```diff
@@ -9,9 +9,32 @@
  * @returns {boolean}
  */
 function wantsKeys (request) {
-  return request.query.keys === 'true'
+  return request.query.keys === 'true' || request.query.keys === true
+}
+
+/**
+ * Returns whether or not to perform a signin verification email.
+ * Currently, the uid is converted to a number between 0-100 which corresponds to the
+ * sign-in confirmation sample rate.
+ *
+ * @param uid
+ * @returns {boolean}
+ */
+function shouldEnableSigninConfirmation(uid, config) {
+  // Sign-in confirmation is determined by whether the feature is enabled and
+  // if the user's first two uid digits fall within the config sample rate.
+  if (config.signinConfirmation && config.signinConfirmation.enabled) {
+
+    // Returns the first 2 integers of uid
+    var uidNum = parseInt(uid.substr(0, 4), 16) % 100
```

Comment:
Just read that back to myself and it sounds like rubbish now I think about it. Ignore me, sorry. 😊